griptape/tools/base_google_client.py

```diff
@@ -1,7 +1,52 @@
 from abc import ABC
 from attr import define, field
 from griptape.tools import BaseTool
+from typing import Optional, Any
+
 
 @define
 class BaseGoogleClient(BaseTool, ABC):
     service_account_credentials: dict = field(kw_only=True)
+
+    def _build_client(self, scopes: list[str], service_name: str, version: str, owner_email: str) -> Any:
+        from google.oauth2 import service_account
+        from googleapiclient.discovery import build
+
+        credentials = service_account.Credentials.from_service_account_info(
+            self.service_account_credentials, scopes=scopes
+        )
+
+        return build(
+            serviceName=service_name,
+            version=version,
+            credentials=credentials.with_subject(owner_email)
+        )
+
+    def _path_to_file_id(self, service, path: str) -> Optional[str]:
```

Comment:
Missing type hint for `service`.

Reply:
Can we rename it to `_convert_path_to_file_id`? Otherwise, it feels like this method returns a "path" to some "file ID," which doesn't make sense :)

Reply:
Makes sense, changed it !